drivers/gpio/gpio_rv32m1.c

```diff
@@ -409,6 +409,8 @@ DEVICE_AND_API_INIT(gpio_rv32m1_porte, DT_OPENISA_RV32M1_GPIO_GPIO_E_LABEL,
 
 static int gpio_rv32m1_porte_init(struct device *dev)
 {
+	CLOCK_EnableClock(kCLOCK_Rgpio1);
```

Comment:
This is interesting.  I think we have an initialization priority issue here.  We have this setup in DTS to work with the PCC driver:
https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/riscv32/rv32m1_ri5cy.dtsi#L197
(note: `clocks = <&pcc1 0x3c>;`)

So in theory this clock should be enabled.

Reply:
To be specific:
PCC1 instance is set to: `PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_OBJECTS`
GPIOE is to: `POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT`

This looks ok at first glance.

Reply:
Wouldn't that require a call to clock_control_driver_api.on() from the GPIO driver to actually be enabled?

Reply:
Wow, it looks like this was totally missed in the gpio driver.   The other ports don't require a GPIO clock to be enabled.  Can you try the following patch to see if it works?
```
diff --git a/drivers/gpio/gpio_rv32m1.c b/drivers/gpio/gpio_rv32m1.c
index 7d16e01932..90500eef6b 100644
--- a/drivers/gpio/gpio_rv32m1.c
+++ b/drivers/gpio/gpio_rv32m1.c
@@ -12,6 +12,7 @@
 #include <soc.h>
 #include <fsl_common.h>
 #include <fsl_port.h>
+#include <clock_control.h>
 
 #include "gpio_utils.h"
 
@@ -409,16 +410,33 @@ DEVICE_AND_API_INIT(gpio_rv32m1_porte, DT_OPENISA_RV32M1_GPIO_GPIO_E_LABEL,
 
 static int gpio_rv32m1_porte_init(struct device *dev)
 {
+	int ret = 0;
+
+#if defined(DT_OPENISA_RV32M1_GPIO_GPIO_E_CLOCK_CONTROLLER)
+	/* enable gpio clock */
+	struct device *clk = device_get_binding(
+				DT_OPENISA_RV32M1_GPIO_GPIO_E_CLOCK_CONTROLLER);
+	if (!clk) {
+		return -ENODEV;
+	}
+
+	ret = clock_control_on(clk, (clock_control_subsys_t)
+				    DT_OPENISA_RV32M1_GPIO_GPIO_E_CLOCK_NAME);
+	if (ret < 0) {
+		return ret;
+	}
+#endif
+
 #ifdef DT_OPENISA_RV32M1_GPIO_GPIO_E_IRQ
 	IRQ_CONNECT(DT_OPENISA_RV32M1_GPIO_GPIO_E_IRQ,
 		    DT_OPENISA_RV32M1_GPIO_GPIO_E_IRQ_PRIORITY,
 		    gpio_rv32m1_port_isr, DEVICE_GET(gpio_rv32m1_porte), 0);
 
 	irq_enable(DT_OPENISA_RV32M1_GPIO_GPIO_E_IRQ);
-
-	return 0;
 #else
-	return -1;
+	ret = -1;
 #endif
+
+	return ret;
 }
 #endif /* CONFIG_GPIO_RV32M1_PORTE */
diff --git a/dts/bindings/gpio/openisa,rv32m1-gpio.yaml b/dts/bindings/gpio/openisa,rv32m1-gpio.yaml
index d06c53f286..dbbe5f8d1f 100644
--- a/dts/bindings/gpio/openisa,rv32m1-gpio.yaml
+++ b/dts/bindings/gpio/openisa,rv32m1-gpio.yaml
@@ -31,6 +31,12 @@ properties:
       description: Human readable string describing the device (used by Zephyr for API name)
       generation: define
 
+    clocks:
+      type: array
+      category: optional
+      description: Clock gate control information
+      generation: define
+
 "#cells":
   - pin
   - flags
```

Reply:
In theory we would probably want the same #if defs for ports A-D, but let's see if this works first.

Reply:
@mike-scott Added and tested. Works.
Would you like me to add similar defines for ports A-D even though they have no clocks?

Reply:
@MaureenHelm @mbolivar Do you think Vega needs clock handling code for ports A-D?  The current HW doesn't need it, but I'm not sure if this code will ever be re-used on another board which needs clocks enabled.

Reply:
The SoC hardware reference manual is a little confusing about whether GPIO instances A-D have clock gates. Table 28-5 on page 885 suggests that they do have clock gates, but the PCC register map in chapter 30 only shows GPIOE having a clock gate. I'm inclined to believe the latter, meaning that A-D do not have clock gates. That said, I think a more robust solution would be to refactor the gpio driver to have a common init function that configures the clock gate if one is available. Something like this:
https://github.com/zephyrproject-rtos/zephyr/blob/6a739435a07c2e25cf9ed5da3bf4c95d15fed25a/drivers/spi/spi_mcux_dspi.c#L273-L278
Where if the dts doesn't specify a clock, then `clock_name` is set to NULL.

Reply:
@MaureenHelm Thanks for the feedback. I have implemented you proposed solution.